client/collection_graph.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type CollectionGraph struct {
+	Revision int                          `json:"revision"`
+	Groups   map[string]map[string]string `json:"groups"`
+}
+
+func (c *Client) GetCollectionGraph() (CollectionGraph, error) {
+	url := fmt.Sprintf("%s/api/collection/graph", c.BaseURL)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+
+	collectionGraph := CollectionGraph{}
+	if err != nil {
+		return collectionGraph, err
+	}
+	if err := c.sendRequest(req, &collectionGraph); err != nil {
+		return collectionGraph, err
+	}
+
+	log.Printf("[DEBUG] Got collection graph '%+v'", collectionGraph)
+	return collectionGraph, nil
+}
+
+func (c *Client) UpdateCollectionGraph(cg CollectionGraph) (CollectionGraph, error) {
+	url := fmt.Sprintf("%s/api/collection/graph", c.BaseURL)
+
+	retries := 10
+	updated := CollectionGraph{}
+	var err_rr error
+	for retries >= 0 {
+		current_cg, err := c.GetCollectionGraph()
+		if err != nil {
+			return cg, err
+		}
+
+		cg.Revision = current_cg.Revision
+		b := new(bytes.Buffer)
+		_ = json.NewEncoder(b).Encode(cg)
+		req, err := http.NewRequest(http.MethodPut, url, b)
+		req.Header.Set("Content-Type", "application/json")
+		if err != nil {
+			err_rr = err
+			break
+		}
+		if err := c.sendRequest(req, &updated); err != nil {
+			if strings.Contains(err.Error(), "collection_revision_pkey") || strings.Contains(err.Error(), "status code: 409") {
+				log.Printf("[ERROR] There were an error with the collection graph, retries left %d.", retries)
+				time.Sleep(time.Duration(rand.Float32()*2) * time.Second)
```

Comment:
Why `*2`? 
I see that `rand.Float32()` returns a value between `[0,1)`. Do you mean to randomly wait for 2x of that time before retry? 
Is this arbitrary or did you observe that retrying faster than that results in errors?